pkg/controller/backingstore/backingstore_controller.go

```diff
@@ -66,3 +68,19 @@ func Add(mgr manager.Manager) error {
 
 	return nil
 }
+
+// ModeChangedPredicate will only allow events that changed Status.Mode.ModeCode.
+// This predicate should be used only for BackingsStore objects!
+type ModeChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Update implements the update event trap for LabelsChangedPredicate
+func (p ModeChangedPredicate) Update(e event.UpdateEvent) bool {
+	if e.ObjectOld == nil || e.ObjectNew == nil {
+		return false
+	}
+	oldBackingStore := e.ObjectOld.(*nbv1.BackingStore)
+	newBackingStore := e.ObjectNew.(*nbv1.BackingStore)
```

Comment:
You should check the ok return value for the cast in order to validate a successful cast and return false if not

Reply:
I don't think that's necessarily a good idea - let me explain:

The current form of type-assertion (with just one ret value) is simpler and more assertive, because if it fails it will cause a runtime panic. Since most of our code is using concrete types (compared to say generic libraries) then a type assert mismatch is not a real functional error that requires any status to the user etc, but it is just a bug and the best thing we could do is make sure we put a spotlight on the issue for devs to easily see the issue.

Reply:
Agreed, resolving.